File: AEDS/BASES/TAD's Sequencial/pilhaS.java
public class pilhaS {
    public class Pilha {
        private int[] array;
        private int n;
        public Pilha (int tam) {
            if (tam <= 0) throw new IllegalArgumentException("Erro!");
            array = new int[tam];
            n = 0;
        }

        public void inserir (int num) {
            if (n >= array.length) throw new RuntimeException("Erro!");
            array[n++] = num;
        }

        public int remover () {
            if (n == 0) throw new RuntimeException("Erro!");
            return array[--n];
        }

        public boolean vazia () { return n == 0; }

        public int[] impPar () {
            int pares = 0, impares = 0;
            for (int i = 0; i < n; i++) {
                if (array[i] % 2 == 0) pares++;
                else impares++;
            }
            int[] result = new int[2];
            result[0] = pares; result[1] = impares;
            return result;
        }

        public void mostrar () {
            System.out.print("[ ");
            for (int i = 0; i < n; i++) System.out.print(array[i] + " ");
            System.out.println("]");
        }

        public void mostrarRec(int i) {
            if (i == n - 1) System.out.print("[ ");
            if (i >= 0) {
                System.out.print(array[i] + " ");
                mostrarRec(i - 1);
            } else System.out.println("]");
        }

        public int topo () {
            if (n == 0) throw new RuntimeException("Erro!");
            return array[n - 1];
        }

        public int tamanho () { return n; }
    }
}

File: AEDS/BASES/TAD's Sequencial/pilhaS.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

#define MAX_TAM 100

int array[MAX_TAM];
int n = 0;

void inserir (int x) {
    if (n >= MAX_TAM) exit(1);
    array[n++] = x;
}

int remover () {
    if (n == 0) exit(1);
    return array[--n];
}

bool vazia () { return n == 0; }

int topo () {
    if (n == 0) exit(1);
    return array[n - 1];
}

int *impPar () {
    int pares = 0, impares = 0;
    for (int i = 0; i < n; i++) {
        if (array[i] % 2 == 0) pares++;
        else impares++;
    }
    int *resultado = (int*)malloc(2*sizeof(int));
    resultado[0] = pares;
    resultado[1] = impares;
    return resultado;
}

void mostrar () {
    printf("[ ");
    for (int i = 0; i < n; i++) printf("%d ", array[i]);
    printf("]\n");
}
void mostrarRec (int i) {
    if (i == n - 1) printf("[ ");
    if (i >= 0) {
        printf("%d ", array[i]);
        mostrarRec(i - 1);
    } else printf("]\n");
}
